Midichlorians/test.cc

```diff
@@ -0,0 +1,135 @@
+/*******************************************************************************
+The MIT License (MIT)
+
+Copyright (c) 2015 Maxim Zhurovich <zhurovich@gmail.com>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*******************************************************************************/
+
+// Use the define below to enable debug output via `NSLog`.
+// #define CURRENT_APPLE_ENABLE_NSLOG
+
+#include "../port.h"
+
+#ifdef CURRENT_APPLE
+
+// The following line is needed to avoid OS X headers conflicts with C++.
+#define __ASSERT_MACROS_DEFINE_VERSIONS_WITHOUT_UNDERSCORES 0
+
+#include "Dev/Beta/iOS/Midichlorians.mm"
+#include "Dev/Beta/iOS/MidichloriansImpl.mm"
+
+#include "../Blocks/HTTP/api.h"
+
+#include "../Bricks/strings/join.h"
+#include "../Bricks/template/rtti_dynamic_call.h"
+
+#include "../Bricks/dflags/dflags.h"
+#include "../3rdparty/gtest/gtest-main-with-dflags.h"
+
+DEFINE_int32(http_port, 8383, "Port to spawn server on.");
+DEFINE_string(http_route, "/log", "HTTP route of the server.");
+
+struct Server {
+  using T_EVENTS = TypeList<iOSAppLaunchEvent, iOSIdentifyEvent, iOSFocusEvent, iOSGenericEvent>;
+
+  Server(int http_port, const std::string& http_route)
+      : messages_processed(0u), http_port(http_port), http_route(http_route) {
```

Comment:
We haz scoped registerers! https://github.com/c5t/Current/blob/master/examples/EmbeddedDB/db.cc#L63-L64